airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py

```diff
@@ -0,0 +1,429 @@
+                self.logger.error(f"Unsuported GA type: {field_type}")
+        except KeyError:
+            attr_type = None
+            self.logger.error(f"Unsuported GA {field_type}: {attribute}")
+
+        data_type = self.map_type.get(attr_type, "string")
+
+        return data_type
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        json_response = response.json()
+        reports = json_response.get(self.report_field, [])
+
+        for report in reports:
+            column_header = report.get("columnHeader", {})
+            dimension_headers = column_header.get("dimensions", [])
+            metric_headers = column_header.get("metricHeader", {}).get("metricHeaderEntries", [])
+
+            for row in self.get_data(report):
+                record = {}
+                dimensions = row.get("dimensions", [])
+                metrics = row.get("metrics", [])
+
+                for header, dimension in zip(dimension_headers, dimensions):
+                    data_type = self.lookup_data_type("dimension", header)
+
+                    if data_type == "integer":
+                        value = int(dimension)
+                    elif data_type == "number":
+                        value = float(dimension)
+                    else:
+                        value = dimension
+
+                    record[header.replace("ga:", "ga_")] = value
+
+                for i, values in enumerate(metrics):
+                    for metric_header, value in zip(metric_headers, values.get("values")):
+                        metric_name = metric_header.get("name")
+                        metric_type = self.lookup_data_type("metric", metric_name)
+
+                        if metric_type == "integer":
+                            value = int(value)
+                        elif metric_type == "number":
+                            value = float(value)
+
+                        record[metric_name.replace("ga:", "ga_")] = value
+
+                yield record
+
+
+class GoogleAnalyticsV4ObjectsBase(GoogleAnalyticsV4Stream):
+    """
+    Main class for all the GoogleAnalyticsV4 data streams (GoogleAnalyticsV4 Object names),
+    provides functionality for dynamically created classes as streams of data.
+    """
+
+    @property
+    def state_checkpoint_interval(self) -> int:
+        return self.window_in_days
+
+    @staticmethod
+    def to_datetime_str(date: datetime) -> str:
+        """
+        Custom method.
+        Returns the formated datetime string.
+        :: Output example: '2021-07-15 07' FORMAT : "%Y-%m-%d"
+        """
+        return date.strftime("%Y-%m-%d")
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        """
+        Override get_json_schema CDK method to retrieve the schema information for GoogleAnalyticsV4 Object dynamically.
+        """
+
+        report = next((ga_stream for ga_stream in self.ga_streams if ga_stream["name"] == self.name), None)
```

Comment:
it should be in __init__ method

Reply:
ga_streams - is not needed in this class at all , we should directly pass init 'report' attr